src/Symfony/Bundle/FrameworkBundle/Logger/Logger.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Logger;
+
+use Psr\Log\AbstractLogger;
+use Psr\Log\InvalidArgumentException;
+use Psr\Log\LogLevel;
+
+/**
+ * Minimalist PSR-3 logger designed to write in stdout, stderr or any other stream.
+ *
+ * @internal
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class Logger extends AbstractLogger
+{
+    private static $defaultOutputByLevel = array(
+        LogLevel::EMERGENCY => 'php://stderr',
+        LogLevel::ALERT => 'php://stderr',
+        LogLevel::CRITICAL => 'php://stderr',
+        LogLevel::ERROR => 'php://stderr',
+        LogLevel::WARNING => 'php://stderr',
+        LogLevel::NOTICE => false,
+        LogLevel::INFO => false,
+        LogLevel::DEBUG => false,
+    );
+    private $outputByLevel;
+    private $handles = array();
+    private $formatter;
+
+    public function __construct(array $outputByLevel = array(), callable $formatter = null)
```

Comment:
Why are you still allowing a formatter callable here, which is never used when wiring the service ?

Reply:
https://github.com/symfony/symfony/pull/24300#discussion_r140799505